core/templates/domain/story/story-node.model.spec.ts

```diff
@@ -318,4 +318,21 @@ describe('Story node model', () => {
 
     expect(_sampleStoryNode.getThumbnailBgColor()).toEqual('#fff');
   });
+
+  it('should return false when nodeId is not a string', () => {
+    /*
+    Using @ts-ignore comment to ignore typescript error and
+    test return value to be false if node id is not a string.
+    */
+    // @ts-ignore
+    expect(_sampleStoryNode._checkValidNodeId({})).toEqual(false);
+    // @ts-ignore
+    expect(_sampleStoryNode._checkValidNodeId([])).toEqual(false);
+    // @ts-ignore
+    expect(_sampleStoryNode._checkValidNodeId(undefined)).toEqual(false);
+    // @ts-ignore
+    expect(_sampleStoryNode._checkValidNodeId(true)).toEqual(false);
+    // @ts-ignore
+    expect(_sampleStoryNode._checkValidNodeId(23)).toEqual(false);
+  });
```

Comment:
The major part of this file is already covered in #14187.
I have just worked on one test in which Soumyajyoti was struggling.
In order to test that `typeof nodeId` is not `string`, we need to pass something else other than `string` in `_checkValidNodeId` function which will always give us a TS error. To bypass the error for these lines I have used
 `// @ts-ignore`. I have also added a comment for the same.